app/scripts/modules/kubernetes/v2/serverGroup/details.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { IController, module } from 'angular';
+
+import {
+  Application,
+  CONFIRMATION_MODAL_SERVICE,
+  IServerGroup,
+  SERVER_GROUP_READER,
+  SERVER_GROUP_WARNING_MESSAGE_SERVICE,
+  SERVER_GROUP_WRITER,
+  ServerGroupReader
+} from '@spinnaker/core';
+
+import { IKubernetesServerGroup } from './IKubernetesServerGroup';
+
+interface IServerGroupFromStateParams {
+  accountId: string;
+  region: string;
+  name: string;
+}
+
+class KubernetesServerGroupDetailsController implements IController {
+  public state = { loading: true };
+  public serverGroup: IKubernetesServerGroup;
+
+  constructor(serverGroup: IServerGroupFromStateParams,
+              public app: Application,
+              private serverGroupReader: ServerGroupReader) {
+    'ngInject';
+
+    this.app
+      .ready()
+      .then(() => this.extractServerGroup(serverGroup))
+      .catch(() => this.autoClose());
+  }
+
+  public canResizeServerGroup(): boolean {
+    return false; // will change with daemon set
+  }
+
+  private autoClose(): void {
+    return;
+  }
+
+  private transformServerGroup(serverGroupDetails: IServerGroup): IKubernetesServerGroup {
+    const serverGroup: IKubernetesServerGroup = Object.assign(serverGroupDetails);
+    const [apiVersion, kind, name] = serverGroup.name.split('|');
+    serverGroup.displayName = name;
+    serverGroup.kind = kind;
+    serverGroup.apiVersion = apiVersion.replace('.', '/');
+    return serverGroup;
+  }
+
+  private extractServerGroup(fromParams: IServerGroupFromStateParams): ng.IPromise<void> {
+    return this.serverGroupReader
+      .getServerGroup(this.app.name, fromParams.accountId, fromParams.region, encodeURIComponent(fromParams.name))
```

Comment:
Is `encodeURIComponent` necessary here? (I have no idea). I've never had to do this - I thought Angular's `$http` service handled this for me, but maybe not.

Reply:
I had an issue earlier with slashes that I fixed by changing the encoding in clouddriver, let's see how it works w/o i